### ISSUE TITLE
[ENH] `predict_proba` prototype with tensorflow_probability

### ISSUE THREAD

**Issue by fkiraly** [alan-turing-institute/sktime#2087]
Experimental prototype for `predict_proba` default for probability distribution return.

This should include:
* mechanics on how to handle soft dependency *within* individual estimators
* design of sensible default behaviour
* possibly, compatible loss functions to showcase interaction with evaluation/scoring

**Comment by fkiraly** [comment]
Done, this is now included in `BaseForecaster` via https://github.com/alan-turing-institute/sktime/pull/2100
